src/ferry_planner/schedule.py

```diff
@@ -254,36 +308,180 @@ def _refresh_task(self) -> None:
+        index = get_seasonal_schedule_daterange_index(hrefs, date)
+        if index < 0:
+            pass  # date is out of range
+        else:
+            url = response.url.scheme + "://" + response.url.host + hrefs[index]
+            if index == 0 or url == str(response.url):
+                rows = get_seasonal_schedule_rows(soup, date)
+            else:
+                result.redirect_url = url
+                return result
+    result.sailings = parse_sailings_from_html_rows(rows, date)
+    if result.sailings is None:
+        for note in [
+            "Seasonal schedules have not been posted for these dates",
+            "Schedules for your selected date and route are currently unavailable",
+        ]:
+            if note in html:
+                result.add_note(note)
+                result.sailings = []
+                return result
+        print(f"No sailings found at {response.url}")
+    return result
+
+
+def parse_sailings_from_html_rows(rows: Sequence[Tag] | None, date: datetime) -> Sequence[FerrySailing] | None:
+    if rows is None:
+        return None
     sailing_row_min_td_count = 3
-    for row in table.tbody.find_all("tr"):
+    sailings: Sequence[FerrySailing] = []
+    notes = None
+    for row in rows:
         tds = row.find_all("td")
-        if len(tds) < sailing_row_min_td_count:
+        if (
+            len(tds) < sailing_row_min_td_count
+            or "No sailings available" in tds[1].text
+            or "No passengers permitted" in tds[1].text
+        ):
             continue
+        td1 = tds[1].text.strip().split("\n", maxsplit=1)
+        if len(td1) > 1:
+            notes = parse_sailig_comment(td1[1])
+            # assumiing dates are always in the first note
+            if is_schedule_excluded_on_date(notes[0], date):
+                continue
+            notes = [n for n in notes if n]
         departure = datetime.strptime(
-            row.find_all("td")[1].text,
+            td1[0].strip(),
             "%I:%M %p",
         ).replace(year=date.year, month=date.month, day=date.day)
         arrival = datetime.strptime(
-            row.find_all("td")[2].text,
+            row.find_all("td")[2].text.strip(),
             "%I:%M %p",
         ).replace(year=date.year, month=date.month, day=date.day)
         td3 = tds[3].text.strip()
+        td3format = "%Hh %Mm" if "h " in td3 and "m" in td3 else "%Mm" if "m" in td3 else "%Hh"
         duration = int(
             datetime_to_timedelta(
                 datetime.strptime(
                     td3,
-                    "%Hh %Mm",
+                    td3format,
                 ),
             ).total_seconds(),
         )
         sailing = FerrySailing(
             departure=departure,
             arrival=arrival,
             duration=duration,
+            notes=notes,
         )
-        yield sailing
+        sailings.append(sailing)
+    return sailings
+
+
+def parse_sailig_comment(comment: str) -> list[str]:
```

Comment:
```suggestion
def parse_sailing_comment(comment: str) -> list[str]:
```
Maybe also refactor as a generator?

Reply:
not in this case, as it will complicate things - first part should contain 'only/except on' part, and remaining notes after that, it is easier to do it in list